### PR TITLE
Fix annotations

### PR DESCRIPTION
**Pull request by azidar**
Fix bug where calling state.copy(circuit = newCircuit) would duplicate annotations.
Also changed getMyAnnotations to return Seq[Annotation]

### CODE REVIEW

File: src/main/scala/firrtl/passes/memlib/ReplaceMemMacros.scala [hunk 1]
```diff
@@ -219,12 +219,12 @@ class ReplaceMemMacros(writer: ConfWriter) extends Transform {
...
-          case PinAnnotation(c, pin) => pin
-          case _ => error(s"Bad Annotations: ${p.values}")
-        }
-      case None => "pin"
+      case Nil => "pin"
+      case Seq(p) => p match {
+        case PinAnnotation(c, pin) => pin
+        case _ => error(s"Bad Annotation: ${p}")
+      }
+      case _ => error("Shouldn't be here")
```
Comment (jackkoenig) [inline src/main/scala/firrtl/passes/memlib/ReplaceMemMacros.scala]: throwInternalError?

File: src/main/scala/firrtl/passes/memlib/ReplaceMemTransform.scala [hunk 2]
```diff
@@ -96,8 +96,9 @@ Optional Arguments:
 class SimpleTransform(p: Pass, form: CircuitForm) extends Transform {
   def inputForm = form
   def outputForm = form
-  def execute(state: CircuitState): CircuitState = state.copy(circuit = p.run(state.circuit))
+  def execute(state: CircuitState): CircuitState = state.copy(circuit = p.run(state.circuit), annotations = None)
```
Comment (jackkoenig) [inline src/main/scala/firrtl/passes/memlib/ReplaceMemTransform.scala]: copy here will still copy the rename map though right? Perhaps it should just be

```scala
def execute(state: CircuitState): CircuitState = CircuitState(p.run(state.circuit), outputForm)
```

File: src/main/scala/firrtl/passes/wiring/WiringTransform.scala [hunk 3]
```diff
@@ -84,6 +85,5 @@ class WiringTransform extends Transform with SimpleRun {
           state.copy(circuit = runPasses(state.circuit, passSeq(winfo)))
         case _ => error("Wrong number of sources, tops, or sinks!")
       }
```
Comment (jackkoenig) [inline src/main/scala/firrtl/passes/wiring/WiringTransform.scala]: I believe there is an implicit match so you can just write:
```scala
p.foreach {
  case SinkAnnotation(...
```

File: src/main/scala/firrtl/Annotations.scala [hunk 4]
```diff
@@ -247,8 +247,8 @@ object Annotations {
           Tuple2(pNMap, ptransformMap)
         }
       }
-    def get(id: Class[_]): Option[Map[Named, Annotation]] = idMap.get(id)
-    def get(named: Named): Option[Map[Class[_], Annotation]] = namedMap.get(named)
+    def get(id: Class[_]): Seq[Annotation] = annotations.filter(a => a.transform == id)
```
Comment (azidar) [inline src/main/scala/firrtl/Annotations.scala]: Yeah, I think we should cross that bridge when we get there. Another idea is for the transforms overwrite getMyAnnotations to pass a different class.

File: src/main/scala/firrtl/passes/memlib/ReplaceMemTransform.scala (hunk 2 above)
Comment (azidar) [inline src/main/scala/firrtl/passes/memlib/ReplaceMemTransform.scala]: The rename map is generally not passed to a new transform, so state.renamemap should always be none. But you are right, we should probably play it safe and just return CircuitState(circuit, form)

File: src/main/scala/firrtl/passes/wiring/WiringTransform.scala (hunk 3 above)
Comment (azidar) [inline src/main/scala/firrtl/passes/wiring/WiringTransform.scala]: done

File: src/main/scala/firrtl/passes/memlib/ReplaceMemMacros.scala (hunk 1 above)
Comment (azidar) [inline src/main/scala/firrtl/passes/memlib/ReplaceMemMacros.scala]: done

**Pull request merged**